src/sentry/eventstore/snuba_discover/backend.py

```diff
@@ -0,0 +1,106 @@
+from __future__ import absolute_import
+
+import six
+from copy import deepcopy
+
+from sentry.utils import snuba
+from sentry.eventstore.base import EventStorage
+from sentry.eventstore.snuba.backend import (
+    ASC_ORDERING,
+    DESC_ORDERING,
+    get_after_event_condition,
+    get_before_event_condition,
+    SnubaEventStorage,
+)
+
+
+class SnubaDiscoverEventStorage(EventStorage):
+    """
+    Experimental backend that uses the Snuba Discover dataset instead of Events
+    or Transactions directly.
+    """
+
+    def get_events(self, *args, **kwargs):
+        return SnubaEventStorage().get_events(*args, **kwargs)
+
+    def get_event_by_id(self, *args, **kwargs):
+        return SnubaEventStorage().get_event_by_id(*args, **kwargs)
+
+    def get_earliest_event_id(self, event, filter):
+        filter = deepcopy(filter)
+        filter.conditions = filter.conditions or []
+        filter.conditions.extend(get_before_event_condition(event))
+        filter.end = event.datetime
+
+        return self.__get_event_id_from_filter(filter=filter, orderby=ASC_ORDERING)
+
+    def get_latest_event_id(self, event, filter):
+        filter = deepcopy(filter)
+        filter.conditions = filter.conditions or []
+        filter.conditions.extend(get_after_event_condition(event))
+        filter.start = event.datetime
+
+        return self.__get_event_id_from_filter(filter=filter, orderby=DESC_ORDERING)
+
+    def get_next_event_id(self, event, filter):
+        """
+        Returns (project_id, event_id) of a next event given a current event
+        and any filters/conditions. Returns None if no next event is found.
+        """
+        assert filter, "You must provide a filter"
+
+        if not event:
+            return None
+
+        filter = deepcopy(filter)
+        filter.conditions = filter.conditions or []
+        filter.conditions.extend(get_after_event_condition(event))
+        filter.start = event.datetime
+
+        return self.__get_event_id_from_filter(filter=filter, orderby=ASC_ORDERING)
+
+    def get_prev_event_id(self, event, filter):
+        """
+        Returns (project_id, event_id) of a previous event given a current event
+        and a filter. Returns None if no previous event is found.
+        """
+        assert filter, "You must provide a filter"
+
+        if not event:
+            return None
+
+        filter = deepcopy(filter)
+        filter.conditions = filter.conditions or []
+        filter.conditions.extend(get_before_event_condition(event))
+        filter.end = event.datetime
+
+        return self.__get_event_id_from_filter(filter=filter, orderby=DESC_ORDERING)
+
+    def __get_columns(self, additional_columns):
+        columns = EventStorage.minimal_columns
+
+        if additional_columns:
+            columns = set(columns + additional_columns)
+
+        return [col.value.event_name for col in columns]
```

Comment:
It seems this is basically copy paste from SnubaEventStorage and the only difference is the dataset name. 
Can you move this in the parent class and customize only the code that actually runs the query unless you are planning to make changes here.

Reply:
Removed `__get_columns` since it's not used. I can't reference the methods from the other class here without actually changing the original implementation, and I kind of wanted to avoid doing that for the purpose of running this short lived test.